src/YieldDistributor.sol

```diff
@@ -140,32 +140,18 @@ contract YieldDistributor is IYieldDistributor, OwnableUpgradeable {
         Checkpoints.Checkpoint208 memory _currentCheckpoint = _sourceContract.checkpoints(_account, 0);
         if (_currentCheckpoint._key > _end) return 0;
 
-        /// Find the latest checkpoint that is within the interval
-        do {
-            --_currentCheckpointIndex;
-            _currentCheckpoint = _sourceContract.checkpoints(_account, _currentCheckpointIndex);
-        } while (_currentCheckpoint._key > _end);
-
-        /// Initialize voting power with the latest checkpoint thats within the interval (or nearest to it)
-        uint48 _latestKey = _currentCheckpoint._key < _start ? uint48(_start) : _currentCheckpoint._key;
-        uint256 _totalVotingPower = _currentCheckpoint._value * (_end - _latestKey);
-
-        if (_latestKey == _start) return _totalVotingPower;
+        uint256 _totalVotingPower;
 
         for (uint32 i = _currentCheckpointIndex; i > 0;) {
```

Comment:
with this refactor moving the logic into a single for loop, I think the variable `_currentCheckpointIndex` would be better renamed to `_numCheckpoints` and then when setting us this for loop you can instead do:

```
for (uint32 i = _numCheckpoints - 1; i > 0; --i)` {
  _currentCheckpoint = _sourceContract.checkpoints(_account, i);
  ...
}
```

Reply:
addressed with 9f92e21